### ISSUE TITLE
Proposal: Add a machine readable form of equivalent words

### ISSUE THREAD

**Issue by goneall** [spdx/license-list-XML#927]
The license matching guidelines list a number of words which are equivalent in the documentation.  Adding this in a machine readable form in the license-XML repository would allow for tools to pick up the latest and greatest with minimal effort.

**Comment by goneall** [comment]
There are several alternatives to the format and naming for the file.

To keep it simple, yet well defined, I would propose:
- A file named "equivalentwords.txt" at the root of the repo
- The file be a UTF-8 encoded file
- The format of the file is the 2 equivalent words separated by a comma on separate lines
- Line endings are single LF characters (Linux style)

For example:
```
acknowledgment,acknowledgement
analogue,analog
analyse,analyze
```

**Comment by swinslow** [comment]
@goneall Makes sense to me! I'll resist the urge to over-engineer this, I think the simple text file approach you described sounds good  :)

**Comment by swinslow** [comment]
This seemed straightforward so I went ahead and did it  :)  @goneall, let me know what you think.

This was based on the list of words from the current matching guidelines at https://spdx.org/spdx-license-list/matching-guidelines.

A couple of notes to add to the format details listed above:
* these have all been converted to lowercase
* they are _not_ fully alphabetized, instead I've maintained the ordering from the matching guidelines page
* note that there is some overlap in terms: -- separate lines for `sublicense,sub-license` and `sub-license,sub license`

For the latter point, I wonder if these should be grouped into a single line?

`sublicense,sub-license,sub license`

Seems clearer to me, but it would mean that there could be >1 comma per line.

**Comment by swinslow** [comment]
@goneall Do you want to give #929 a quick review? If you approve then I can go ahead and merge it, if in the next couple days then can include it for the 3.7 release.

**Comment by goneall** [comment]
> For the latter point, I wonder if these should be grouped into a single line?

Restricting it to 2 per line makes it a little easier to program to (e.g. can use a simple map), but it is a bit harder to read/maintain by humans.

I'm leaning to to 2 per line but I don't feel strongly about it.

**Comment by zvr** [comment]
I also have no preference, as long as we are all clear in the semantics:

Since we're talking "equivalence",

- the pairs X,Y and Y,Z also cover X,Z
- the same way X,Y also covers Y,X.

Actually, @goneall, I'd think that having all equivalent forms in one line would be easier to program, since there would be no need to check for pre-existing equivalents...

Do we plan to add quotes and hyphens?

**Comment by goneall** [comment]
> I'd think that having all equivalent forms in one line would be easier to program, since there would be no need to check for pre-existing equivalents...

@zvr I agree - I previously programmed the pre-existing equivalents, but if I had to start from scratch having all in the same line would be easier.

> Do we plan to add quotes and hyphens?

There's also end of line characters and spaces if we want to be complete.  Adding this would make it a bit harder to read.  In the Java tools, I use a regex with the UTF codes for most of the the punctuation and special characters, so I probably wouldn't use the comma separated data.

**Comment by jlovejoy** [comment]
so, will we keep the list also in the table in the Matching Guidelines? I'm not a big fan of having the same thing in two places for obvious reasons... Maybe once this is added, we could link to this file from the Matching Guidelines?

As for the 2 items, versus more - it'd look cleaner to allow more than two, but I'm not a programmer, so that's less than two cents worth of input :)

I think it'd be fine (and easier to read) if we alphabetized the list. I can't think of any good reason it is not now.

**Comment by swinslow** [comment]
Hi @goneall @jlovejoy @zvr -- I'm looking back at this and at the draft PR I'd submitted at #927. There were some questions / open items both above and in the #927 discussion, so I'm pulling those together here. Let me know if you disagree with any of my proposed actions below.

* **here or in a spec appendix?** -- I think the preference is that this list of equivalent words should live in the license-list-XML repo. That way (1) it is easily updateable by the legal team; and (2) it can easily be pulled by tools. If so then also the matching guidelines should be updated to remove the copy of this list and to point to this file instead.

* **at root of repo or in DOCS directory?** -- @goneall had originally suggested a file at the repo root. I tend to agree, if the intention is that this would be pulled by tools, and we can keep DOCS as documentation for humans.

* **separate or combined lines where >2 equivalent?** -- based on comments above it sounds like it would be preferable (for tool programmers) if e.g. "sublicense" and "sub-license" and "sub license" were all on one line, rather than spread on multiple lines.

* **alphabetize?** -- list should be alphabetized

Unless there's any objection, I can update my draft PR with these changes and then we can merge it.

**Comment by goneall** [comment]
@swinslow Agree with all your proposals above.  Thanks for working on this!

**Comment by swinslow** [comment]
Just updated the PR to include the changes above. I did _not_ update the matching guidelines in this same PR, because they are still living on the HTML site (with HTML copy in DOCS directory).

I'll open a separate issue to track so that after #929 is merged, I'll update the matching guidelines to point to this file.

**Comment by swinslow** [comment]
Given the OKs above, I will go ahead and merge #929 as well as #941. Will track website update in #969. Thanks all!

**Comment by swinslow** [comment]
Sorry -- to clarify, will not be merging #941 as-is (since DOCS/matching-guidelines is just a mirror of the website), but will update website in #969 and then backport the changes to reflect in this repo.
